allennlp/data/dataset_readers/huggingface_datasets_reader.py

```diff
@@ -0,0 +1,276 @@
+            if isinstance(value, ClassLabel):
+                fields_to_be_added[feature] = LabelField(
+                    inputs[1][feature], label_namespace=feature, skip_indexing=True
+                )
+
+            # datasets Value can be of different types
+            elif isinstance(value, Value):
+
+                # String value maps to TextField
+                if value.dtype == "string":
+                    # datasets.Value[string] maps to TextField
+                    # If tokenizer is provided we will use it to split it to tokens
+                    # Else put whole text as a single token
+                    if self.tokenizer is not None:
+                        fields_to_be_added[feature] = TextField(
+                            self.tokenizer.tokenize(inputs[1][feature])
+                        )
+
+                    else:
+                        fields_to_be_added[feature] = TextField([Token(inputs[1][feature])])
+
+                else:
+                    fields_to_be_added[feature] = LabelField(
+                        inputs[1][feature], label_namespace=feature, skip_indexing=True
+                    )
+
+            elif isinstance(value, Sequence):
+                # We do not know if the string is token or text, we will assume text and make each a TextField
+                # datasets.features.Sequence of strings maps to ListField of TextField
+                if hasattr(value.feature, "dtype") and value.feature.dtype == "string":
+                    field_list2: typing.List[TextField] = list()
+                    for item in inputs[1][feature]:
+                        # If tokenizer is provided we will use it to split it to tokens
+                        # Else put whole text as a single token
+                        tokens: typing.List[Token]
+                        if self.tokenizer is not None:
+                            tokens = self.tokenizer.tokenize(item)
+
+                        else:
+                            tokens = [Token(item)]
+
+                        item_field = TextField(tokens)
+                        field_list2.append(item_field)
+
+                    fields_to_be_added[feature] = ListField(field_list2)
+
+                # datasets Sequence of strings to ListField of LabelField
+                elif isinstance(value.feature, ClassLabel):
+                    field_list = list()
+                    for item in inputs[1][feature]:
+                        item_field = LabelField(
+                            label=item, label_namespace=feature, skip_indexing=True
+                        )
+                        field_list.append(item_field)
+
+                    fields_to_be_added[feature] = ListField(field_list)
+
+                else:
+                    self.raise_feature_not_supported_value_error(value)
+
+            # datasets.Translation cannot be mapped directly
+            # but it's dict structure can be mapped to a ListField of 2 ListField
```

Comment:
This is not accurate anymore, right? Because it now creates two fields with different names?